### ISSUE TITLE
How can I get predicted values from neural network on test data during training.

### ISSUE THREAD

**Issue by Da-Bean** [rstudio/keras#1219]
Hi,

I want to get predicted values on test data during training without saving the weights.
I just want to get that on each epoch using the prediction function.
Because it is too hard to save every weight matrix corresponding epoch.

Thanks,

**Comment by t-kalinowski** [comment]
When posting for help, it's helpful to try to boil down your issue to a small reproducible example. Ideally someone should be able to copy and paste a code snippet into their terminal and reproduce the output you have questions about.

If I understand your question correctly, you want to call `predict` on the test set at the end of each epoch during training, and then do something custom with the return value of the `predict` call, like save it. I think the easiest way to do this will be to define a custom callback using `callback_lambda`, and then pass that call back to the `fit()` call. Please take a look at this article: https://keras.rstudio.com/articles/training_callbacks.html
